docs/04-api-reference/front-commerce-core/graphql.mdx

```diff
@@ -0,0 +1,435 @@
+import { AcmeQuery } from "~/graphql/graphql";
+
+const AcmeGraphQLApi = createGraphQLApi(request, user, config);
+
+const response = await AcmeGraphQLApi(AcmeQuery);
+```
+
+## `makeErrorLoggerInterceptor`
+
+Intercept and log errors
+
+```js
+makeErrorLoggerInterceptor({
+  isMaintenance: maintenance,
+  ignoreForbiddenErrors: ignore,
+});
+```
+
+Arguments:
+
+- `maintenance` (`function`): a function that recieve error as parameter and
+  return a `boolean` to tell if website is in maintenance mode
+- `ignore` (`boolean`): a boolean to tell if forbidden errors should be ignored
+
+_Example_
+
+```ts
+import { makeErrorLoggerInterceptor } from "@front-commerce/core/graphql";
+
+const errorLoggerInterceptor = makeErrorLoggerInterceptor({
+  isMaintenance: (error) =>
+    error.response &&
+    (error.response.status === 503 ||
+      (error.response.status === 500 &&
+        /maintenance mode .* enabled/.test(error.response.data))),
+  ignoreForbiddenErrors: ignoreForbiddenErrors,
+});
+```
+
+## `limitRateByClientIp`
+
+Limit a GraphQL mutation by client IP
+
+```ts
+limitRateByClientIp(config, resolver);
+```
+
+Arguments:
+
+- `config` (`object`): rate limiter configuration object:
+  - `max` (`number`): number of calls allowed within window
+  - `window` (`number`): window of maximum calls
+- `resolver` (`function`): GraphQL resolver
+
+_Example_
+
+```ts title="extension/acme/graphql/modules/contact/resolver.ts"
+import { limitRateByClientIp } from "@front-commerce/core/graphql";
+
+export default {
+  Mutation: {
+    sendContactEmail: limitRateByClientIp(
+      { max: 5, window: "1m" },
+      (_, { template, data, storeEmail }, { loaders }) =>
+        loaders.Contact.sendContactEmail(template, data, storeEmail)
+    ),
+  },
+};
+```
+
+## `reorderForIds`
+
+Reorder an array by ids
+
+```ts
+reorderForIds(ids, (key = "id"));
+```
+
+Arguments:
+
+- `ids` (`string[]` default to `id`): ids to reorder
+- `key` (`string`): key to use to order
+
+_Example_
+
+```ts
+import { reorderForIds } from "@front-commerce/core/graphql";
+
+const orderer = reorderForIds([2, 3, 1], "id");
+const reordered = orderer([{ id: 1 }, { id: 2 }, { id: 3 }]);
+```
+
+## `reorderForIdsCaseInsensitive`
+
+Same as [`reorderForIds`](#) but case insensitive
+
+## `makeBatchLoaderFromSingleFetch`
+
+Convert a single fetch to a batch loader
+
+```ts
+makeBatchLoaderFromSingleFetch(fetch, response);
+```
+
+Argumvents:
+
+- `fetch` (`function`): fetch function
+- `response` (`object`): response transformer
+
+_Example_
+
+```ts
+import { makeBatchLoaderFromSingleFetch } from "@front-commerce/core/graphql";
+
+const axiosInstance = axios.create({
+  baseURL: "https://acme.com/api",
+});
+
+const acmeLoader = makeBatchLoaderFromSingleFetch(
+  (depth) => axiosInstance.get(`/prodcuts`),
+  (response) => response.data.products
+);
+```
+
+## `maybeValue`
+
+Check if a value is defined, else print a debug message and return `null`
```

Comment:
```suggestion
Checks if a value is defined, else prints a debug message and returns `null`.
```

**thought:** I'm not sure if we should be talking about debug prints, or if we do, I think we must also document to what debug "key" it is linked (here, `DEBUG=axios`).